src/charm.py

```diff
@@ -24,22 +28,80 @@ def __init__(self, *args: typing.Any) -> None:
             args: passthrough to CharmBase.
         """
         super().__init__(*args)
+        self._charm_state = CharmState(charm_config=self.config)
```

Comment:
I'd prefer passing the config elements the class needs and if there are too many, a named tuple as transitional object

Reply:
In my opinion, passing every individual configuration option breaks boundary between the `FlaskCharm` and `CharmState`. The process of determining which configuration options make up the charm state should be an internal implementation detail of CharmState, separated from the concerns of FlaskCharm.

Reply:
In my opinion the state should only know and get the information it needs, passing the whole charm config doesn't make sense for that as state will have information it doesn't need, doesn't care and shouldn't care.
In terms of design boundaries any object should not get access to any data or method it's not suppose to read or call (hence the concept of interface that only exposes what you need to expose).

@jdkandersson a topic to discuss together

Reply:
If the configuration is passed the the `CharmState` one by one. When the `CharmState` changed, for example, a new configuration was added, the interface of the `CharmState` has to change too to add the new configuration, this breaks the open-close principle.

Reply:
Well it does make sense if the charmstate changes, it requires changes to its interface if that information must be exposed / consumed.
After discussing with David we came to a proposal that has been reflected in the charm complexity spec see:
https://docs.google.com/document/d/1G62PosrObvmQY5KbxvqaxByojlhDxrmNtcbPS39YbaY/edit#bookmark=id.y9g4mcrurrid

Reply:
I am updating the CharmState to accept individual charm configuration values instead of the entire configuration. However, I have encountered another problem in the operator framework testing module. The charm class is initialized only once for the entire test case. So, when handling multiple events in unit test cases, the charm constructor is called only once. This causes CharmState to become stalled with old data since the data update happens in the charm class constructor. I am not sure whether this is a bug or a feature.